Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp

```diff
@@ -252,6 +370,45 @@ namespace Multiplayer
         m_drawParams.m_position.SetY(m_drawParams.m_position.GetY() - textHeight - m_lineSpacing);
     }
 
+    void MultiplayerConnectionViewportMessageSystemComponent::DrawScrim(float alphaMultiplier) const
+    {
+        AZ::RPI::ViewportContextPtr viewport = AZ::RPI::ViewportContextRequests::Get()->GetDefaultViewportContext();
+        if (!viewport)
+        {
+            return;
+        }
+
+        AzFramework::DebugDisplayRequestBus::BusPtr debugDisplayBus;
+        AzFramework::DebugDisplayRequestBus::Bind(debugDisplayBus, viewport->GetId());
+        AzFramework::DebugDisplayRequests*  debugDisplay = AzFramework::DebugDisplayRequestBus::FindFirstHandler(debugDisplayBus);
+        if (!debugDisplay)
+        {
+            return;
+        }
+
+        const AZ::u32 previousState = debugDisplay->GetState();
+        debugDisplay->SetState(
+            e_Mode2D | e_AlphaBlended | e_FillModeSolid | e_CullModeBack | e_DepthWriteOff | static_cast<AZ::u32>(e_DepthTestOff));
```

Comment:
Other than 2d mode these can all be done using the direct access functions. 2D mode would require a Draw2DQuadGradient function.
```c++
debugDisplay->SetAlpha(ScrimAlpha * alphaMultiplier);
// fill mode unsupported
debugDisplay->CullOn();
debugDisplay->DepthWriteOff();
debugDisplay->DepthTestOff();
```

Reply:
That would be nice. In other words, add a new DebugDisplayRequests method: DrawQuadGradient2d?

Reply:
Added DrawQuad2dGradient to DebugDisplayRequests. Let me know what you think!